Assets/Mirror/Runtime/Authenticator.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Unity Event for the NetworkConnection
+    /// </summary>
+    [Serializable] public class UnityEventNetworkConnection : UnityEvent<NetworkConnection> { }
+
+    /// <summary>
+    /// Base class for implementing component-based authentication during the Connect phase
+    /// </summary>
+    [HelpURL("https://mirror-networking.com/xmldocs/articles/Concepts/Authentication.html")]
+    public abstract class Authenticator : MonoBehaviour
```

Comment:
why was this renamed?
I think NetworkAuthenticator.cs was a great name, fits well with NetworkManager.cs and it's often on the same GameObject too :)

Reply:
Because it's abstract now, like Transport.   The actual components we make, including the Basic I am shipping, do/will have Add Component and a submenu of Network Authenticators.

Reply:
please rename to NetworkAuthenticator again, so we can get this merged.
NetworkBehaviour is abstract too, but we use that naming convention almost everywhere in Mirror :)

Reply:
Done, but you will never see Network Authenticator in the inspector.